include/devicetree/pwms.h

```diff
@@ -0,0 +1,383 @@
+ * @param inst instance number
+ * @param idx logical index into the property
+ * @return the label property for the named specifier at index idx
+ */
+#define DT_INST_PWMS_LABEL_BY_IDX(inst, idx) \
+	DT_PWMS_LABEL_BY_IDX(DT_DRV_INST(inst), idx)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller "name" (label property) by name
+ * (see @ref DT_PWMS_LABEL_BY_NAME)
+ * @param inst instance number
+ * @param name lowercase-and-underscores specifier name
+ * @return the label property for the named specifier by name
+ */
+#define DT_INST_PWMS_LABEL_BY_NAME(inst, name) \
+	DT_PWMS_LABEL_BY_NAME(DT_DRV_INST(inst), name)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller "name"
+ * (see @ref DT_PWMS_LABEL_BY_IDX)
+ * @param inst instance number
+ * @return the label property for the named specifier at index 0
+ */
+#define DT_INST_PWMS_LABEL(inst) DT_INST_PWMS_LABEL_BY_IDX(inst, 0)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller "cell" value at an index
+ * @param inst instance number
+ * @param cell binding's cell name within the specifier at index "idx"
+ * @param idx logical index into the property
+ * @return the value of the cell inside the specifier at index "idx"
+ */
+#define DT_INST_PWMS_CELL_BY_IDX(inst, cell, idx) \
+	DT_PWMS_CELL_BY_IDX(DT_DRV_INST(inst), cell, idx)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller "cell" value by name
+ * (see @ref DT_PWMS_CELL_BY_NAME)
+ * @param inst instance number
+ * @param name lowercase-and-underscores specifier name
+ * @param cell binding's cell name within the specifier referenced as "name"
+ * @return the value of the cell inside the named specifier
+ */
+#define DT_INST_PWMS_CELL_BY_NAME(inst, name, cell) \
+	DT_PWMS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller "cell" value at an index 0
+ * @param inst instance number
+ * @param cell binding's cell name within the specifier at index 0
+ * @return the value of the cell inside the specifier at index 0
+ */
+#define DT_INST_PWMS_CELL(inst, cell) \
+	DT_INST_PWMS_CELL_BY_IDX(inst, cell, 0)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller 'channel' at an index
+ * @param inst instance number
+ * @param idx logical index into the property
+ * @return the channel value for the named specifier at index idx
+ * @see DT_PHA_BY_IDX()
+ * @see DT_PHA()
+ */
+#define DT_INST_PWMS_CHANNEL_BY_IDX(inst, idx) \
+	DT_INST_PWMS_CELL_BY_IDX(inst, channel, idx)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller 'channel' by name
+ * @param inst instance number
+ * @param name lowercase-and-underscores "pwm" name
+ * @return the channel value for the named specifier by name
+ * @see DT_INST_PWMS_CELL_BY_NAME()
+ */
+#define DT_INST_PWMS_CHANNEL_BY_NAME(inst, name) \
+	DT_INST_PWMS_CELL_BY_NAME(inst, name, channel)
+
+/**
+ * @brief Get a DT_DRV_COMPAT pwm controller 'channel' value at an index 0
+ * @param inst instance number
+ * @return the pin value for the named specifier at index idx
```

Comment:
pin copy/paste issue

Reply:
fixed

Reply:
No named specifier here either, since it's accessed by index, not name